### PR TITLE
Adding case temperature controller

### PR DESCRIPTION
**Pull request by AnMW**

### CODE REVIEW

File: firmware/include/caseTempController/config.h [hunk 1]
```diff
@@ -0,0 +1,12 @@
+//*****************************************//
+// configuration                           //
+//*****************************************//
+#define K_P				1	    // proportional factor of the PID controller
+#define K_I				0.1  	// integral factor of the PID controller
+#define K_D				0.5	    // derivative factor of the PID controller
+#define TAU				20	    // number of measurements viewed in calculation, reduce to save space and calculation times
+#define GOAL_TEMP		30	    // goal temperature of 30°C
+#define TEMP_GPIO		4	    // the GPIO data pin of the temperature sensor
+#define FAN_GPIO	    16      // the PWM pin for fan control
```
Comment (thegreatdane6) [inline firmware/include/caseTempController/config.h:10]: pls check with with @felixherbst to match the korrekt pins

File: firmware/src/caseTempController/fan.cpp [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+// includes                                //
+//*****************************************//
+#include "caseTempController/fan.h"
+
+
+//*****************************************//
+// defines                                 //
+//*****************************************//
+#define FREQUENCY   25000   // the PWM frequency (fan runs best at 25kHz)
+#define RESOLUTION  8       // the duty cycle resolution in bits (8 means 255 steps)
```
Comment (thegreatdane6) [inline firmware/src/caseTempController/fan.cpp:11]: how about expanding the existing fan stuff and adding another func to control the utility fan?

File: firmware/src/main.cpp [hunk 3]
```diff
@@ -222,4 +228,7 @@ void loop()
 
   // handlet the current rotary position
   rotary.handle();
+
+  // regulate case temperature
+  caseTempController.caseTempController();
```
Comment (thegreatdane6) [inline firmware/src/main.cpp:233]: Should this function not be little bit more selfdescribing? like "readTemperatureAndAdjustFan" or something

File: firmware/include/caseTempController.h [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        Fan fan;
+
+        // counter used to check if controller should start measuring/calculating
+        int sleep_check;
+
+    public:
+
+        CaseTempController()
+        {
+            this->sleep_check = 0;
```
Comment (thegreatdane6) [inline firmware/include/caseTempController.h:34]: What does this do?

File: firmware/src/main.cpp (hunk 3 above)
Comment (AnMW) [reply firmware/src/main.cpp:233]: Can do

File: firmware/include/caseTempController/config.h (hunk 1 above)
Comment (thegreatdane6) [reply firmware/include/caseTempController/config.h:10]: pin 26 ist der temp sensor
pin 23 ist der utility fan

Comment (AnMW) [reply firmware/include/caseTempController/config.h:10]: Thank you, I'm changing it.

File: firmware/src/caseTempController/fan.cpp (hunk 2 above)
Comment (AnMW) [reply firmware/src/caseTempController/fan.cpp:11]: I don't want to rip apart the current structure where everything concerning the controller is in one directory, so I'm kind of not feeling good with this. If you insist I will change it, though.

File: firmware/include/caseTempController.h (hunk 4 above)
Comment (AnMW) [reply firmware/include/caseTempController.h:34]: I do not want to measure/adjust on every loop-through of the main file. Instead, I'm doing so on every 30000th loop in order to save some calculation time (and not trash your logs too much).

In the function readTemperatureAndAdjustFanSpeed (formerly known as caseTempController), the whole logic is encased by an "if (sleep_check % SLEEP == 0)", where SLEEP can be found in config.h and is 30000.

File: firmware/include/caseTempController/config.h (hunk 1 above)
Comment (mlkiefer) [reply firmware/include/caseTempController/config.h]: Wouldn't it be better to have all pin assignments in one file instead of several?

Comment (thegreatdane6) [reply firmware/include/caseTempController/config.h]: I can live with the way it currently is

Comment (thegreatdane6) [reply firmware/include/caseTempController/config.h]: Maybe we can Chance this in the future though

Comment (AnMW) [reply firmware/include/caseTempController/config.h]: @mlkiefer You/We can open up an issue for this if you want to.

**Review by thegreatdane6 — APPROVED** [review]

**Pull request merged**